File: matlab/eom_z_derivation.m
syms m k g r_T r_0 real
syms z(t) r(t)
syms z_0 zdot_0

% Define the variable r(t)
r = (t/T)*(r_T - r_0) + r_0;

% Define the second-order differential equation
eq = m*diff(z, t, 2) == k*(r - z) - m*g
% Display the solution using a formula
figure;
annotation('textbox', [0.1 0.1 0.8 0.8], 'String', ['$$' latex(eq) '$$'], ...
    'Interpreter', 'latex', 'FontSize', 14, 'FitBoxToText', 'on');
axis off;

% Solve the differential equation with initial conditions
sol = dsolve(eq);
% Solve the differential equation with initial conditions
% sol = dsolve(eq, z(0) == z_0, subs(diff(z, t), t, 0) == zdot_0);

% Display the solution
disp('Symbolic Solution:');
latexString = latex(sol);
disp(latexString);

% Display the solution using a formula
figure;
annotation('textbox', [0.1 0.1 0.8 0.8], 'String', ['$$' latexString '$$'], ...
    'Interpreter', 'latex', 'FontSize', 14, 'FitBoxToText', 'on');
axis off;

% Save LaTeX-formatted solution to a file
fid = fopen('slip_eom_z_latex_solution.tex', 'w');
fprintf(fid, '%s', latex(sol));
fclose(fid);

File: python/robust_physics_based_eom_derivation/eom_z_derivation.py
from sympy import Function, dsolve, Eq, Derivative, symbols, latex

# Define the variables and functions
t, T, r_T, r_0, k, m, g, z0, zdot0 = symbols('t T r_T r_0 k m g z0 zdot0')
z = Function('z')
r = (t * (r_T - r_0) / T) + r_0

# Define the differential equation
ode = Eq(m * Derivative(z(t), t, t), k * (r - z(t)) - m * g)

# Define the initial conditions as symbolic expressions
initial_conditions = {z(0): z0, Derivative(z(t), t).subs(t, 0): zdot0}

# Solve the differential equation
solution = dsolve(ode, z(t), ics=initial_conditions)

# Display the LaTeX-formatted solution
latex_solution = latex(solution)
print(latex_solution)
